Add example using CMakeToolchain with Ninja as generator

uilianries opened the pull request:
Hi!

In this new entry, I tried to explain how to build the Conan template project `cmake_exe` using CMake + Ninja.

My initial idea was describing a little about CMake generators, then how to use Conan configuration in different ways.

Current Preview:

![Screenshot 2024-06-11 at 11-02-04 Using CMakeToolchain with different generators Ninja example — conan 2 4 0 documentation](https://github.com/conan-io/docs/assets/4870173/5ffb8479-5905-4ec3-8717-72661f19a00e)

closes #3767

RubenRBS commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,104 @@
+
+Understanding CMake Generators
+------------------------------
+CMake offers a variety of `generators <https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html>`_ to create build system files. If you want to use a generator other than the default chosen by CMake, you can configure ``tools.cmake.cmaketoolchain:generator``.
```
I'd add a note here to clarify the user not to get consued with Conan's generators (`CMakeToolchain` etc), and CMake's (`Ninja` etc)

czoido commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,104 @@
+
+    $ cmake --help
+
+You can set this configuration in your profile, directly in the command line, or even in your global configuration.
+
```
maybe some links here would be nice?

uilianries replied:
I added in footnotes, I'll move to the documentation, to be more clear.

uilianries replied:
Done.

czoido commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,104 @@
+
+You can set this configuration in your profile, directly in the command line, or even in your global configuration.
+
+Generating the Toolchain File with the Ninja Generator
+------------------------------------------------------
```
I think we can remove this title?

uilianries replied:
Removed.

czoido commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,104 @@
+------------------------------------------------------
+The project's recipe declares the "CMakeToolchain" generator.
+
+1. **Default Build**: To build the recipe using the default generator chosen by CMake, run:
+
+    .. code-block:: bash
+
+        $ conan create .
+
```
Maybe obviate the default case?

uilianries replied:
Moved as a step before building.

czoido commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,104 @@
+    build_type=Release
+
+    [conf]
+    *:tools.cmake.cmaketoolchain:generator=Ninja
```
why the * if that's not needed if you want to apply yo everything?
```suggestion
    tools.cmake.cmaketoolchain:generator=Ninja
```

uilianries replied:
Removed.

memsharded commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,104 @@
+
+.. code-block:: bash
+
+    $ conan new -d name=foo -d version=0.1.0 cmake_exe
```
Positional argument first, please, as we recommend ourselves
```suggestion
    $ conan new cmake_exe -d name=foo -d version=0.1.0
```

uilianries replied:
Ops! I'll update the other reference too: https://docs.conan.io/2/examples/tools/cmake/cmake_toolchain/build_project_cmake_presets.html#cmaketoolchain-building-your-project-using-cmakepresets

uilianries replied:
Done, fixed all usages for `cmake new `

czoido commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,117 @@
+
+The Ninja generator will be used by default when running CMake with this profile.
+
+Setting the Ninja Generator Globally
```
I think, that instead of having a section for each way of using this configuration, I would pick the one we think it's more appropriate for this conf (maybe setting using a profile) and then just mentioning or just redirecting to the docs on other sections...

uilianries replied:
Right, let's add to profile as example and add other as reference.

Plus, the CMake helper supports hardcoded generator.

uilianries commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,104 @@
+This guide demonstrates how to use ``CMakeToolchain`` with predefined generators like ``Ninja`` and how to configure it to use different generators.
+
+Creating a Basic Project
+-------------------------
```
Simplified to a single title now: Using the Ninja generator by default in a profile

uilianries commented:
@czoido I've updated, it's simpler now. No longer shows example with global and command line, but use profile by default. There are links exposing how to configure it using others. 

@RubenRBS Please, take a look in the cmake generators section. I added links and a note to avoid confusion.

@memsharded The `cmake new` usage is now fixed and I checked other parts of the documentation using the same pattern.

memsharded commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,79 @@
+-------------------------------------------------
+
+To set the Ninja generator as the default in a profile file,
+add the entry ``[conf]`` to your profile with the generator value:
```
It this is a generic ``myprofile`` then put it a ``caption`` "myprofile", and then use ``conan create . -pr=myprofile`` below.

If this change is proposed to the default profile, say it here: ``add entry to your *default* profile "<cache>/profiles/default" ...``

uilianries replied:
@memsharded I update the example by using a custom profile.

czoido commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,79 @@
+    foo/0.1.0: Running CMake.configure()
+    foo/0.1.0: RUN: cmake -G "Ninja" ...
+
+Conclusion
+----------
+
+This guide showed you how to configure `CMakeToolchain` to use different generators, specifically the Ninja generator,
+through the command line, profile configuration, and global settings.
```
The conclusion is not adding much, maybe we can cut it?
```suggestion
Conclusion
----------

This guide showed you how to configure `CMakeToolchain` to use different generators, specifically the Ninja generator,
through the command line, profile configuration, and global settings.
```

memsharded commented on examples/tools/cmake/cmake_toolchain/use_different_toolchain_generator.rst:
```diff
@@ -0,0 +1,84 @@
+=============================================================
+
+This guide demonstrates how to use :ref:`CMakeToolchain<conan_tools_cmaketoolchain>` with predefined generators
+like `Ninja <https://ninja-build.org/>`_ and how to configure it to use different generators.
```
Add something like this: This example assumes that Ninja is installed in your system. If it is not, you can use ninja from a Conan package with tool_requires (maybe add an example of this at the end of the example?)

uilianries replied:
@memsharded I added a section about installing Ninja as platform_tool_requires: https://github.com/conan-io/docs/pull/3771/commits/10264d5b0938e370117b950f6118d45b0af26f80

Please, tell me if it's the recommended way.

uilianries replied:
Done, using tool requires instead of platform tool requires.

memsharded reviewed (approved):
Looking good

uilianries commented:
The preview in the description is updated, in case you want to have a view of the rendered version.

RubenRBS reviewed (approved):
VP of wording approves 👍 , let's see what @czoido has to say about the contents themselves

uilianries commented:
Thank you everyone!
